packages/db/tests/test-executions.js

```diff
@@ -0,0 +1,53 @@
+const test = require('ava');
+const cryptoRandomString = require('crypto-random-string');
+
+const { removeNilProperties } = require('@cumulus/common/util');
+const { translateApiExecutionToPostgresExecution } = require('../dist/executions');
+
+test('translateApiExecutionToPostgresExecution converts API execution to Postgres', async (t) => {
+  const now = Date.now();
+
+  const apiExecution = {
+    arn: 'arn:aws:lambda:us-east-1:1234:1234',
+    name: `${cryptoRandomString({ length: 10 })}execution`,
+    execution: 'https://test',
+    error: { test: 'error' },
+    tasks: {},
+    type: 'IngestGranule',
+    status: 'running',
+    createdAt: now,
+    updatedAt: now,
+    timestamp: now,
+    originalPayload: { testInput: 'originalPayloadValue' },
+    finalPayload: { testOutput: 'finalPayloadValue' },
+    duration: 2,
+  };
+
+  // Note that we are not testing the foreign keys here. Properties like parent_cumulus_id,
+  // collection_cumulus_id, and async_operation_cumulus_id are set in
+  // translateApiExecutionToPostgresExecution using helpers that are tested elsewhere.
+  // The complete execution, including those foreign keys is
+  // tested in the data-migration2 integration tests.
```

Comment:
👋. Open to testing the foreign keys here. I didn't because it requires bringing in and setting up Knex to actually test if the cumulus_ids are picked up correctly. It also requires things like the DynamoDB models for Collections and Async Operations to be imported here. We're doing this in the Lambda's `test-executions.js`.

If it bothers anyone that there are properties not tested in this file, I can definitely look at how to mock some of the cumulus_id stuff.

Reply:
FWIW, you could do a minor refactor of translateApiExecutionToPostgresExecution so that the PG models are optional arguments to the function. Then you could pass "fake" PG models to the tests that return cumulus IDs for the records, so you can test that this function properly assigns cumulus ID values to the record when they're received from the database. The actual logic fetching cumulus IDs from the database is already tested elsewhere, so we don't need to test it again here

For example

https://github.com/nasa/cumulus/blob/feature-rds/packages/db/src/translate/rules.ts#L20
https://github.com/nasa/cumulus/blob/feature-rds/packages/db/tests/translate/test-rules.js#L4

Reply:
Oh right! You showed this pattern in a group review a while back. This is exactly the use case for passing mock models around.

I've updated this to:
```
export const translateApiExecutionToPostgresExecution = async (
  dynamoRecord: ExecutionRecord,
  knex: Knex,
  collectionPgModel = new CollectionPgModel(),
  asyncOperationPgModel = new AsyncOperationPgModel(),
  executionPgModel = new ExecutionPgModel()
)
```
All of the Postgres models are instantiated in the function arguments, rather than the function body. I'm still creating an instance of `Logger` in the body because it's not really something we want to mock in the same way. The lesson I'm adopting here is something like "Always accept data models as arguments and set defaults to a new instance of that model for easier testing."

Reply:
Yeah, I don't know if I would say **always**. I'm sure there are some people who would question this approach, but it seems like a sane way to allow for easy mocking to me